crates/fuel-core/src/state/in_memory/memory_store.rs

```diff
@@ -129,6 +129,18 @@ where
     ) -> BoxedIter<KVItem> {
         self.iter_all(column, prefix, start, direction).into_boxed()
     }
+
+    fn iter_store_keys(
+        &self,
+        column: Self::Column,
+        prefix: Option<&[u8]>,
+        start: Option<&[u8]>,
+        direction: IterDirection,
+    ) -> BoxedIter<fuel_core_storage::kv_store::KeyItem> {
+        self.iter_all(column, prefix, start, direction)
```

Comment:
[nit] We could avoid `value.clone()` if we implemented that in some another way=) Maybe we could use generics  to minimize the code duplciation

Reply:
yeah, maybe we should change the fn signature of `iter_all_keys` in `IteratorOverTable` to include the column, prefix, start key. it is currently - 
```
    fn iter_all_keys<M>(
        &self,
        direction: Option<IterDirection>,
    ) -> BoxedIter<super::Result<M::OwnedKey>>
    where
        M: Mappable,
        Self: IterableTable<M>,
    {
        self.iter_all_filtered_keys::<M, [u8; 0]>(None, None, direction)
    
    ```

Reply:
nevermind, fixed in 1665b8606

thanks!

Reply:
will modify `keys_iterator` to do the same